src/hotspot/share/opto/vectornode.hpp

```diff
@@ -1387,32 +1387,19 @@ class VectorMaskCastNode : public VectorNode {
 class VectorReinterpretNode : public VectorNode {
  private:
   const TypeVect* _src_vt;
-  BasicType  _dst_bt;
-  BasicType  _src_bt;
 
  protected:
   uint size_of() const { return sizeof(VectorReinterpretNode); }
  public:
   VectorReinterpretNode(Node* in, const TypeVect* src_vt, const TypeVect* dst_vt)
-      : VectorNode(in, dst_vt), _src_vt(src_vt) {
-     assert(!dst_vt->isa_vectmask() && !src_vt->isa_vectmask(), "");
-     _src_bt = src_vt->element_basic_type();
-     _dst_bt = dst_vt->element_basic_type();
-     init_class_id(Class_VectorReinterpret);
-  }
-
-  VectorReinterpretNode(Node* in, BasicType src_bt, const TypeVect* src_vt,
-                        BasicType dst_bt, const TypeVect* dst_vt)
-      : VectorNode(in, dst_vt), _src_vt(src_vt) {
-     assert(dst_vt->isa_vectmask() && src_vt->isa_vectmask() || type2aelembytes(src_bt) >= type2aelembytes(dst_bt),
+     : VectorNode(in, dst_vt), _src_vt(src_vt) {
+     assert((!dst_vt->isa_vectmask() && !src_vt->isa_vectmask()) ||
+            (type2aelembytes(src_vt->element_basic_type()) >= type2aelembytes(dst_vt->element_basic_type())),
```

Comment:
Only narrowing re-interpretation is supported if source and destinations are mask type operands i.e.
If Int512Mask is reinterpreted as Byte512Mask each true bit of Int species will correspond to 4 true bits of Byte species, 
but reverse re-interpretation is not straight forwards, since not all 4 bits of byte species may be true and thus cannot be directly reinterpreted as 1 true mask bit for integer species.